### PR TITLE
chore: generate protocol during browser roll

### PR DESCRIPTION
**Pull request by dgozman**
We now commit protocol.ts files during the roll. New `utils/roll_browser.js` helps with that.

This makes our installation very shallow:
- build installer;
- download browsers.

### CODE REVIEW

**Review by pavelfeldman — APPROVED** [review]

**Review by aslushnikov — APPROVED** [review]
Awesome!!

File: .github/workflows/publish_release.yml [hunk 1]
```diff
@@ -19,6 +19,7 @@ jobs:
         registry-url: 'https://registry.npmjs.org'
     - uses: microsoft/playwright-github-action@v1
     - run: npm ci
+    - run: npm run build
```
Comment (aslushnikov) [inline .github/workflows/publish_release.yml:22]: why do we need to do `npm run build` after `npm ci`?

File: utils/roll_browser.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    console.log(`Please specify the browser name, e.g. 'chromium'.`);
+    console.log(`Try running ${SCRIPT_NAME} --help`);
+    process.exit(1);
+  } else if (args.length < 2) {
+    console.log(`Please specify the revision`);
+    console.log(`Try running ${SCRIPT_NAME} --help`);
+    process.exit(1);
+  }
+  const browserName = args[0];
+  if (!['chromium', 'firefox', 'webkit'].includes(browserName)) {
```
Comment (aslushnikov) [inline utils/roll_browser.js:53]: - can we also support `cr`, `ff` and `wk`?
- `browserName.tolowerCase()`?

File: utils/roll_browser.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    console.log(`Unknown browser "${browserName}"`);
+    console.log(`Try running ${SCRIPT_NAME} --help`);
+    process.exit(1);
+  }
+  const revision = args[1];
+
+  // 2. Update browsers.json.
```
Comment (aslushnikov) [inline utils/roll_browser.js:60]: can we switch step 2 and step 3 so that it won't write browsers.json unless it successfully downloaded the browser?

**Review by aslushnikov — APPROVED** [review]

**Pull request merged**